### ISSUE TITLE
Zero-sized channel

### ISSUE THREAD

**Issue by Night-walker** [daokoder/dao#366]
```ruby
(dao) c = mt::Channel<int>(0)
[[Error::Param]] --- Invalid parameter(s):
channel capacity must be greater than 0
Raised by:  Channel.Channel<int>(), from namespace "mt";
```
Since one can set capacity to zero at any time with `.cap(0)`, I don't see necessity for such limitation.

**Comment by daokoder** [comment]
Zero capacity is reserved for closed channels. When you set the capacity to zero, you close the channel.

**Comment by Night-walker** [comment]
Yes, which means that channel capacity *can* be zero, contrary to what the error message claims. One can always change capacity to zero and then to non-zero and etc., so 0 is valid as any other value here.

**Comment by daokoder** [comment]
> Yes, which means that channel capacity can be zero, contrary to what the error message claims. One can always change capacity to zero and then to non-zero and etc., so 0 is valid as any other value here.

Some improvements need to be done for this.
